library/sha256.c

```diff
@@ -56,20 +70,27 @@
 
 #include "mbedtls/platform.h"
 
-#if defined(__aarch64__)
+#if defined(MBEDTLS_ARCH_IS_ARMV8)
 
 #  if defined(MBEDTLS_SHA256_USE_A64_CRYPTO_IF_PRESENT) || \
     defined(MBEDTLS_SHA256_USE_A64_CRYPTO_ONLY)
+#       ifdef __ARM_NEON
+#           include <arm_neon.h>
+#       else
+#           if defined(MBEDTLS_SHA256_USE_A64_CRYPTO_IF_PRESENT)
```

Comment:
Should we rename `MBEDTLS_SHA256_USE_A64_CRYPTO_IF_PRESENT` ? Now it is not only for aarch64.

Reply:
I agree, `MBEDTLS_xxx_USE_ARMV8A_CRYPTO_yyy` would be better. Not sure if we can do this before 4.0, since it would be an API break... obviously we can add a backwards-compatibility section

Reply:
Agreed. I've pushed a commit to do this - see what you think. I can revert it if it's controversial.